src/components/icon-button/icon-button.pw.tsx

```diff
@@ -0,0 +1,215 @@
+    );
+    await expect(iconButton(page)).toBeVisible();
+  });
+
+  test(`should render with a child`, async ({ mount, page }) => {
+    await mount(<IconButtonComponent />);
+
+    await expect(iconButton(page)).toBeVisible();
+  });
+
+  test(`should render with disabled prop`, async ({ mount, page }) => {
+    await mount(<IconButtonComponent disabled />);
+
+    await expect(iconButton(page)).toBeDisabled();
+  });
+});
+
+test.describe("check events for IconButton component", () => {
+  test(`should call onBlur callback when a blur event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <IconButtonComponent
+        onBlur={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await iconButton(page).focus();
+    await iconButton(page).blur();
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should call onFocus callback when a focus event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <IconButtonComponent
+        onFocus={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await iconButton(page).focus();
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should call onMouseEnter callback when a mouseover event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <IconButtonComponent
+        onMouseEnter={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await iconButton(page).hover();
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should call onMouseLeave callback when a mouseout event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <IconButtonComponent
+        onMouseLeave={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await iconButton(page).hover();
+    await page.mouse.move(100, 0);
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should call onClick callback when a click event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <IconButtonComponent
+        onClick={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+    await iconButton(page).click();
+    expect(callbackCount).toBe(1);
+  });
+
+  ["Enter", "Space"].forEach((key) => {
+    test(`should call onClick callback when ${key} key is triggered`, async ({
+      mount,
+      page,
+    }) => {
+      let callbackCount = 0;
+      await mount(
+        <IconButtonComponent
+          onClick={() => {
+            callbackCount += 1;
+          }}
+        />
+      );
+
+      await iconButton(page).press(key);
+      expect(callbackCount).toBe(1);
+    });
+  });
+});
+
+test.describe("check accessibility tests for IconButton component", () => {
+  test(`should render aria-label prop for accessibility`, async ({
```

Comment:
**suggestion**: I  think these test descriptions have been changed and i'm not 100% sure why, I think they should all follow something like `should pass accessibility tests when rendered with an aria-label`, `should pass accessibility tests when rendered with a child`, `should pass accessibility tests when disabled` etc.